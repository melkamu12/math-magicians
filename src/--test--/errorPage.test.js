import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import Errorpage from '../components/error';

describe('Error Page component', () => {
  it('should match DOM Snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Errorpage />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the ErrorPage component correctly', () => {
    render(
      <MemoryRouter>
        <Errorpage />
      </MemoryRouter>,
    );
    const welcome = screen.queryByText('An error occured!');
    expect(welcome).toBeInTheDocument();
  });
});
