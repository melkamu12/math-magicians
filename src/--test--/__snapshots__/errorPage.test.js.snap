// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Page component should match DOM Snapshot 1`] = `
Array [
  <header
    className="header"
  >
    <h1>
      Math Magicians
    </h1>
    <nav>
      <ul
        className="List"
      >
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
          >
            Home |
          </a>
        </li>
        <li>
          <a
            href="/MathCalculator"
            onClick={[Function]}
          >
            Calculator |
          </a>
        </li>
        <li>
          <a
            href="/Quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main>
    <h1>
      An error occured!
    </h1>
    <p>
      Could not find this page
    </p>
  </main>,
]
`;
